doc/console/Portable-Console-Device-High-Level-Design.md

```diff
@@ -0,0 +1,506 @@
+        """
+        Retrives whether portable console device is plugged in or not.
+        This method is mandatory for factory function to auto detect vendor name and model name.
+
+        :return: A boolean, True if portable console device is plugged in
+                          , False otherwise.
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def get_vendor_name(cls):
+        """
+        Retrives the vendor name of the `PortableConsoleDeviceBase` concrete subclass.
+        This method is mandatory for factory function to create instance from manual configuration.
+
+        :return: A string, denoting vendor name of the `PortableConsoleDeviceBase` concrete subclass.
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def get_model_name(cls):
+        """
+        Retrives the model name of the `PortableConsoleDeviceBase` concrete subclass.
+        This method is mandatory for factory function to create instance from manual configuration.
+
+        :return: A string, denoting model name of the `PortableConsoleDeviceBase` concrete subclass.
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def get_instance(cls):
+        """
+        Build and return the portable console device object.
+
+        :return: A tuple of two values.
+                 The first value is an object derived from `PortableConsoleDeviceBase`. If object
+                 cannot be created (eg. due to device not plugged in), return None.
+                 The second value is a string. If object cannot be created, return the error message.
+                 Otherwise, return an empty string.
+        """
+        raise NotImplementedError
+
+    def get_num_devices(self):
+        """
+        Retrieves the number of portable console devices plugged in (by daisy-chain).
+
+        :return: An integer, the number of portable console devices plugged in.
+        """
+        raise NotImplementedError
+
+    def get_serial_numbers(self):
+        """
+        Retrieves the serial numbers of portable console devices.
+
+        :return: A dict, the key is the index of the portable console device (integer, 0-based),
+                         the value is serial number (string).
+                 e.g.
+                 {
+                     0: "device-S/N-1",
+                     1: "device-S/N-2",
+                     ...
+                 }
+        """
+        raise NotImplementedError
+
+    def get_num_lines(self):
+        """
+        Retrieves the number of console lines on portable console devices.
+
+        :return: An integer, the number of console lines on portable console devices.
+        """
+        raise NotImplementedError
+
+    def get_all_lines(self):
+        """
+        Retrieves the infomation of all console lines on portable console devices.
+
+        :return: A dict, the key is console line number (integer, same as portable console device designed),
```

Comment:
If you daisy-chain 2 same console devices, and this model have 8ports. How the port name of 2nd console devices aligned with physical port name?